app/src/main/java/com/github/featuredetectandroid/MainActivity.kt

```diff
@@ -24,4 +55,100 @@ class MainActivity : ComponentActivity() {
             }
         }
     }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        cameraExecutor.shutdown()
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<String>,
+        grantResults:
+            IntArray
+    ) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if (requestCode == REQUEST_CODE_PERMISSIONS) {
+            if (allPermissionsGranted()) {
+                startCamera()
+            } else {
+                Toast.makeText(
+                    this,
+                    "Permissions not granted by the user.",
+                    Toast.LENGTH_SHORT
+                ).show()
+                finish()
+            }
+        }
+    }
+
+    private fun allPermissionsGranted() = REQUIRED_PERMISSIONS.all {
+        ContextCompat.checkSelfPermission(
+            baseContext,
+            it
+        ) == PackageManager.PERMISSION_GRANTED
+    }
+
+    private fun startCamera() {
+        val cameraProviderFuture = ProcessCameraProvider.getInstance(this)
+
+        cameraProviderFuture.addListener({
+            val cameraProvider: ProcessCameraProvider = cameraProviderFuture.get()
+
+            val imageAnalyzer = ImageAnalysis.Builder()
+                .setOutputImageFormat(OUTPUT_IMAGE_FORMAT_YUV_420_888)
+                .build()
+                .also {
+                    it.setAnalyzer(cameraExecutor, PhotoAnalyzer())
+                }
+
+            val cameraSelector = CameraSelector.DEFAULT_BACK_CAMERA
+
+            try {
+                cameraProvider.unbindAll()
+                cameraProvider.bindToLifecycle(
+                    this,
+                    cameraSelector,
+                    imageAnalyzer
+                )
+            } catch (exc: Exception) {
+                Log.e(TAG, "Use case binding failed", exc)
+            }
+        }, ContextCompat.getMainExecutor(this))
+    }
+
+    private class PhotoAnalyzer : ImageAnalysis.Analyzer {
+        var grayScaleByteArray: ByteArray = byteArrayOf()
+        var width: Int = 0
+        var height: Int = 0
+
+        private fun ByteBuffer.toByteArray(): ByteArray {
+            rewind()
+            val data = ByteArray(remaining())
+            get(data)
+            return data
+        }
+
+        override fun analyze(image: ImageProxy) {
+            width = image.width
+            height = image.height
+            val buffer = image.planes[0].buffer
+            grayScaleByteArray = buffer.toByteArray()
+            ViewGrayscale(grayScaleByteArray, width, height)
```

Comment:
I didn't look at the code much as it is WIP and, as I see, the problem lies here.

You can, for example, create a `ViewModel` with `MutableState` fields representing the arguments that are passed here. When created, `ViewGrayscale` will take this `ViewModel` as argument and work with its fields.

Reply:
So, here you'll just set these fields to the values you need